persist/storer.go

```diff
@@ -3,168 +3,222 @@ package persist
 	filePerm        = 0644
 	dirFilePerm     = 0755
 	integrationsDir = "nr-integrations"
 )
 
 var (
-	now = time.Now
+	// ErrNotFound defines an error that will be returned when trying to access a storage entry that can't be found
+	ErrNotFound = errors.New("key not found")
 )
 
-// Errors
-var (
-	ErrNotFound   = errors.New("key not found")
-	ErrTsNotFound = errors.New("timestamp for key not found")
-)
+var now = time.Now
 
-// Storer is a key-value structure that is initialized and stored in a persistent device.
-// It also saves the timestamp when a key was stored.
+// Storer defines the interface of a Key-Value storage system, which is able to store the timestamp
+// where the key was stored.
 type Storer interface {
-	// Set sets the value under the given key, storing the current timestamp that is also returned.
-	// This method does not persist until Save is called.
+	// Set associates a value with a given key. Implementors must save also the time when it has been stored and return it.
+	// The value can be any type.
 	Set(key string, value interface{}) int64
-	// Get returns the value for a given key. Last Set call timestamp is returned.
-	// If key is not found ErrNotFound error is returned.
-	Get(key string) (value interface{}, timestamp int64, err error)
-	// Delete removes the cached data for the given key
-	Delete(key string)
-	// Save persists all in-memory stored data.
+	// Get gets the value associated to a given key and stores in the value referenced by the pointer passed as argument.
+	// It returns the Unix timestamp when the value was stored (in seconds), or an error if the Get operation failed.
+	// It may return any type of value.
+	Get(key string, valuePtr interface{}) (int64, error)
+	// Delete removes the cached data for the given key. If the data does not exist, the system does not return
+	// any error.
+	Delete(key string) error
+	// Save persists all the data in the storer.
 	Save() error
 }
 
+// In-memory implementation of the storer
 type inMemoryStore struct {
-	Data       map[string]interface{}
+	cachedData map[string]jsonEntry
+	Data       map[string][]byte
 	Timestamps map[string]int64
 }
 
+// Holder for any entry in the JSON storage
+type jsonEntry struct {
+	Timestamp int64
+	Value     interface{}
+}
+
+// fileStore is a Storer implementation that uses the file system as persistence backend, storing
+// the objects as JSON.
+// This requires that any object that has to be stored is Marshallable and Unmarshallable.
+// TODO: make this implementation read all the cached values at the same time and persist them at the same time after
+// invoking a Persist() or Save() function. This will improve performance while minimizing disk usage.
 type fileStore struct {
 	inMemoryStore
 	path string
+	ilog log.Logger
 }
 
 // SetNow forces a different "current time" for the Storer.
-// This function is useful only for unit testing.
+// Deprecated: This function is useful only for unit testing outside the persist package.
```

Comment:
`@deprecated` is the standard docblock way

Reply:
I thought the Golang's official, standard way was "Deprecated:"
https://blog.golang.org/godoc-documenting-go-code

Actually, this is the only way IntelliJ IDEA recognizes me the code as deprecated:
![image](https://user-images.githubusercontent.com/939550/40474760-271a4234-5f40-11e8-86ff-b7e3393fb91c.png)

Reply:
I was wrong! thanks for the correction!